x-pack/plugins/security/server/plugin.ts

```diff
@@ -238,36 +253,34 @@ export class Plugin {
       httpResources: core.http.resources,
       logger: this.initializerContext.logger.get('routes'),
       config,
-      authc: this.authc,
-      authz,
+      authz: this.authz,
       license,
       session,
       getFeatures: () =>
         startServicesPromise.then((services) => services.features.getKibanaFeatures()),
       getFeatureUsageService: this.getFeatureUsageService,
+      getAuthenticationService: () => {
+        if (!this.authenticationStart) {
+          throw new Error('Authentication service is not started!');
+        }
+
+        return this.authenticationStart;
+      },
     });
 
-    return deepFreeze<SecurityPluginSetup>({
+    return Object.freeze<SecurityPluginSetup>({
```

Comment:
**question** for my own education, why did we change from a deep freeze to a shallow freeze?

Reply:
I knew you'd ask :see_no_evil: The "trigger" is that I initially wanted to use licensing plugin mock for the `license` contract property in the tests, but the mock uses  `BehaviorSubject` internally that cannot be used with `deepFreeze` (circular references IIRC).

But then I checked and saw that nobody is doing that anyway, so I just kept normal freeze to freeze the top level shape that has a higher chance of being mutated unintentionally.